openassessment/xblock/openassessmentblock.py

```diff
@@ -1242,3 +1242,24 @@ def get_xblock_id(self):
         Returns the xblock id
         """
         return text_type(self.scope_ids.usage_id)
+
+    def index_dictionary(self):
+        """
+        Return dictionary prepared with module content and type for indexing.
+        """
+        # return key/value fields in a Python dict object
+        # values may be numeric / string or dict
+        # default implementation is an empty dict
+        xblock_body = super(OpenAssessmentBlock, self).index_dictionary()
+        content = {
+            "display_name": self.title,
+            "prompt": self.prompt
```

Comment:
Prompt isn't a simple string here, it is a JSON field. You will need to parse it and add all the prompts separately.

Reply:
@xitij2000 It is simple string

```python
    prompt = String(
        default=DEFAULT_PROMPT,
        scope=Scope.content,
        help="The prompts to display to a student."
    )
```
although it can contain html tags which I think I should remove.

**Example**
```python
In [1]: from xmodule.modulestore.django import modulestore                                      
   ...:  
   ...: from opaque_keys.edx.keys import UsageKey                                               
   ...:  
   ...: store = modulestore()                                                                   
   ...:  
   ...: key = UsageKey.from_string("block-v1:ABKUni+CS101+2020_T2+type@openassessment+block@3580
   ...: 9bfd971f4acc9b55e5a7c737b77d") 
   ...: block = store.get_item(key) 
   ...: print(block.index_dictionary()) 
```

```python
{'content_type': 'ORA', 'content': {'prompt': '<p>What is computer?</p>', 'display_name': 'Quiz about computers'}}
```

Reply:
**Update:** If there are multiple prompts then it appears as list of dicts e.g
```python
Prompt: [{"description": "<p>C++, Rust</p>"}, {"description": "<p>Hello World, python, ruby</p>"}]
```

Reply:
@kshitij2000 Yes, you are right :+1: